src/par_iter/test.rs

```diff
@@ -272,6 +272,144 @@ pub fn check_range_exact_and_bounded() {
 }
 
 #[test]
+pub fn check_cmp() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.cmp(&b);
+
+    assert!(result == ::std::cmp::Ordering::Equal);
+}
+
+#[test]
+pub fn check_cmp_lt() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (1..1024).collect();
+
+    let result = a.cmp(&b);
+
+    assert!(result == ::std::cmp::Ordering::Less);
+}
+
+#[test]
+pub fn check_cmp_gt() {
+    let a: Vec<usize> = (1..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.cmp(&b);
+
+    assert!(result == ::std::cmp::Ordering::Greater);
+}
+
+#[test]
+pub fn check_partial_cmp() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.partial_cmp(&b);
+
+    assert!(result == Some(::std::cmp::Ordering::Equal));
+}
+
+#[test]
+pub fn check_partial_cmp_lt() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (1..1024).collect();
+
+    let result = a.partial_cmp(&b);
+
+    assert!(result == Some(::std::cmp::Ordering::Less));
+}
+
+#[test]
+pub fn check_partial_cmp_gt() {
+    let a: Vec<usize> = (1..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.partial_cmp(&b);
+
+    assert!(result == Some(::std::cmp::Ordering::Greater));
+}
+
+#[test]
+pub fn check_partial_cmp_none() {
+    let mut a: Vec<f32> = (0..1024u16).map(From::from).collect();
+    let b: Vec<f32> = (0..1024u16).map(From::from).collect();
+
+    a[0] = ::std::f32::NAN;
+
+    let result = a.partial_cmp(&b);
+
+    assert!(result == None);
+}
+
+#[test]
+pub fn check_eq() {
+    let a: Vec<usize> = (0..1024).collect();
```

Comment:
These tests are actually not testing your code. You are building vectors and then comparing them sequentially. You need something like:

```rust
let a = (0..1024).into_par_iter(); // NB: not collect() :)
let b = (0..1024).into_par_iter();
let result = a.eq(b);
assert!(result);
```

However, my preference is to write the tests so that they compare par against seq iterators, rather than directly encoding the expected result:


```rust
assert_eq!((0..1024).into_par_iter().eq(0..1024), (0..1024).eq(0..1024));
```